.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,123 @@
+name: Build and Publish Images
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+permissions:
+  id-token: write # needed for keyless signing
+  packages: write # needed for ghcr access
+
+env:
+  CONTROLLER: ${{ github.event.repository.name }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Restore Go cache
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
```

Comment:
```suggestion
```
This shouldn't be needed anymore. https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs